source/event_stream_channel_handler.c

```diff
@@ -0,0 +1,535 @@
+            AWS_LOGF_TRACE(AWS_LS_EVENT_STREAM_CHANNEL_HANDLER, "id=%p: processing prelude buffer", (void *)handler);
+
+            struct aws_byte_cursor prelude_cursor = aws_byte_cursor_from_buf(&event_stream_handler->message_buf);
+
+            event_stream_handler->running_crc =
+                aws_checksums_crc32(prelude_cursor.ptr, sizeof(uint32_t) + sizeof(uint32_t), 0);
+            AWS_LOGF_DEBUG(
+                AWS_LS_EVENT_STREAM_CHANNEL_HANDLER,
+                "id=%p: calculated prelude CRC of %" PRIu32,
+                (void *)handler,
+                event_stream_handler->running_crc);
+
+            aws_byte_cursor_read_be32(&prelude_cursor, &event_stream_handler->current_message_len);
+
+            AWS_LOGF_DEBUG(
+                AWS_LS_EVENT_STREAM_CHANNEL_HANDLER,
+                "id=%p: read total message length of %" PRIu32,
+                (void *)handler,
+                event_stream_handler->current_message_len);
+            if (event_stream_handler->current_message_len > AWS_EVENT_STREAM_MAX_MESSAGE_SIZE) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_EVENT_STREAM_CHANNEL_HANDLER,
+                    "id=%p: message length of %" PRIu32 " exceeds the max size of %zu",
+                    (void *)handler,
+                    event_stream_handler->current_message_len,
+                    (size_t)AWS_EVENT_STREAM_MAX_MESSAGE_SIZE);
+                aws_raise_error(AWS_ERROR_EVENT_STREAM_MESSAGE_FIELD_SIZE_EXCEEDED);
+                error_code = aws_last_error();
+                goto finished;
+            }
+
+            /* advance past the headers field since we don't really care about it at this point */
+            aws_byte_cursor_advance(&prelude_cursor, sizeof(uint32_t));
+
+            uint32_t prelude_crc = 0;
+            aws_byte_cursor_read_be32(&prelude_cursor, &prelude_crc);
+            AWS_LOGF_DEBUG(
+                AWS_LS_EVENT_STREAM_CHANNEL_HANDLER,
+                "id=%p: read prelude CRC of %" PRIu32,
+                (void *)handler,
+                prelude_crc);
+
+            /* make sure the checksum matches before processing any further */
+            if (event_stream_handler->running_crc != prelude_crc) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_EVENT_STREAM_CHANNEL_HANDLER,
+                    "id=%p: prelude CRC mismatch. calculated %" PRIu32 " but the crc for the message was %" PRIu32,
+                    (void *)handler,
+                    event_stream_handler->running_crc,
+                    prelude_crc);
+                aws_raise_error(AWS_ERROR_EVENT_STREAM_PRELUDE_CHECKSUM_FAILURE);
+                error_code = aws_last_error();
+                goto finished;
+            }
+        }
+
+        /* read whatever is remaining from the message */
+        if (event_stream_handler->message_buf.len < event_stream_handler->current_message_len) {
```

Comment:
Is there no upper bound on message size, other than 2GB for 32 bit int max?

Reply:
already checked above in this branch:

if (event_stream_handler->current_message_len > AWS_EVENT_STREAM_MAX_MESSAGE_SIZE)